### ISSUE TITLE
bug: editing cluster.yaml doesn't seem to accept dns: or addons: stanzas

### ISSUE THREAD

**Issue by tibers** [rancher/rancher#19110]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

1. Create a cluster. My cluster happens to be on prem so I created the cluster in the UI and then added my nodes via the magic SSH line.
1. Edit the cluster.
1. Add something to the yaml
1. Click Save

That something was:
```
dns:
  provider: coredns
```

I also tried the addons array to a yaml file to install coredns.

**Result:**

The cluster.yaml isn't saved and my DNS provider doesn't become coredns.

**Other details that may be helpful:**

**Environment information**
Rancher 2.1.7
HA installation

**Cluster information**
- Cluster type: Custom
- Machine type: Bare metal, 100GB drive, 32GB RAM
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-03-01T23:34:27Z", GoVersion:"go1.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:30:26Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

RancherOS 1.5.1

```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:20:43 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:28:38 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by loganhz** [comment]
```
dns:
  provider: coredns
```

It should be only available in 2.2

If you want to do it with RKE in 2.1, you can use RKE 0.2.0 to create a cluster and import into Rancher 2.1

**Comment by tibers** [comment]
WELL THEN. Thanks man.
